crates/type-system/src/ontology/property_type/wasm.rs

```diff
@@ -0,0 +1,42 @@
+#![allow(
+    non_snake_case,
+    reason = "We want camelCase types in TS, and this module is WASM only"
+)]
+use wasm_bindgen::prelude::*;
+
+use crate::{PropertyType, ValidationError};
+
+#[wasm_bindgen]
+extern "C" {
+    #[wasm_bindgen(typescript_type = "PropertyType")]
+    pub type IPropertyType;
+}
+
+use tsify::Tsify;
+
+use crate::uri::VersionedUri;
+
+// TODO: this is a temporary solution, we need to figure out error handling across the WASM boundary
+#[derive(Tsify)]
+#[wasm_bindgen]
+pub struct TempError {}
+
+/// Checks if a given {PropertyType} is valid
+///
+/// @throws {TempError} if the property type is malformed
+#[wasm_bindgen]
+pub fn isValidPropertyType(propertyTypeObj: &IPropertyType) -> Result<(), TempError> {
+    propertyTypeObj
+        .into_serde::<PropertyType>()
+        .map_err(|_| TempError {})?;
+    Ok(())
+}
+
+// #[wasm_bindgen]
+// pub fn newPropertyType(
+//     uri: &str,
+//     title: &str,
+//     description: &str,
+// ) -> Result<IPropertyType, TempError> {
+//     let property_type = PropertyType::new(VersionedUri::from_str(uri), title.to_string(), None,
+// ()); }
```

Comment:
For what is this needed/needs to be kept?

Reply:
This is the part I'm currently modifying (that I spoke about offline and is the reason this PR is still _technically_ in draft). The changes should pretty much just be localised to here though so the rest of the review is intact

Reply:
I've actually removed it. I was trying to come up with arguments for `newPropertyType` and I'm not sure how to create a more ergonomic input for the "oneOf" yet that isn't just the same as the normal interface. I'll interface with @nathggns as he starts work on the editors